### PR TITLE
Feature/10314 port request submission on behalf

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Refactor request for on behalf submission of checkins.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10314

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionError.swift [hunk 1]
```diff
@@ -5,102 +5,32 @@
...
 	case notAuthorized
-	case coronaTestServiceError(CoronaTestServiceError)
-
-	/// User has not granted access to their keys
-	case keysNotShared
-
-	/// Access to keys was granted but no keys were collected by the exposure notification framework
-	case noKeysCollected
-
-	case noSubmissionConsent
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionError.swift:20]: where are all these removed errors gone?

**Review by pascal-brause — COMMENTED** [review]
I think you missed to remove the onBehalfCheckinSubmissionURL in the HTTPClient+Configuration?
And the submissionURL you removed in the PR before, right?

File: src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionError.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionError.swift:20]: noSubmissionConsent is now part of ExposureSubmissionServicePreconditionError

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionError.swift:20]: In general: some of them where not used anymore. A lot of them where moved to Resources but not deleted here.

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**